### PR TITLE
Chore: Refactor CLI infrastructure

### PR DESCRIPTION
**Pull request by molant**
This refactor cleans up the current code, making easier to add new
features to the CLI. It also homogenizes how the different actions
are tested.

### CODE REVIEW

File: src/lib/cli.ts [hunk 1]
```diff
@@ -150,90 +61,13 @@ export const execute = async (args: string | Array<string> | Object): Promise<nu
...
-    if (configPathFromInput) {
-        config = Config.load(configPathFromInput);
-    } else {
-        config = await loadOrCreateIfNotExits();
-    }
+    let handled = false;
 
-    if (!config) {
-        logger.log(`Unable to find a valid configuration file. Please add a .sonarrc file by running 'sonar --init'. `);
+    while (cliActions.length > 0 && !handled) {
```
Comment (sarvaje) [inline src/lib/cli.ts]: Some of the actions can throw exceptions. It think it would be nice to have a `try/catch` wrapping the `while`

Comment (molant) [reply src/lib/cli.ts]: Change is pushed. Wrapping individual actions instead of the `while` though.

File: src/lib/cli/rules/delete-core-rule.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    text = text.replace(`\n${itemToDelete}`, '');
+
+    return util.promisify(fs.writeFile)(indexPath, text, 'utf-8');
+};
+
+/** List of questions to ask the user. */
+const questions = [
+    {
+        message: 'What\'s the file name of this rule to be removed? (No file extension needed)',
```
Comment (alrra) [inline src/lib/cli/rules/delete-core-rule.ts]: Use \`What's the file name...\`.

File: src/lib/cli/version.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import * as path from 'path';
+
+
```
Comment (alrra) [inline src/lib/cli/version.ts]: Remove empty lines.

(I'll updated all these minor things before I merge)

File: tests/lib/cli/analyze.ts [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    t.context.config.getFilenameForDirectory.restore();
+    t.context.config.load.restore();
+    t.context.generator.initSonarrc.restore();
+    t.context.spinner.start.restore();
+    t.context.spinner.fail.restore();
+    t.context.spinner.succeed.restore();
+    t.context.inquirer.prompt.restore();
+});
+
+test.serial('if config is not defined, it should get the config file from the directory process.cwd()', async (t) => {
```
Comment (alrra) [inline tests/lib/cli/analyze.ts]: For consistency use `If` instead of `if`, here and everywhere else.

**Pull request closed without merging**